src/Trace/SpanProcessor/ResponseTimeSpanProcessor.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Solarwinds\ApmPhp\Trace\SpanProcessor;
+
+use OpenTelemetry\API\Behavior\LogsMessagesTrait;
+use OpenTelemetry\API\Globals;
+use OpenTelemetry\API\Metrics\HistogramInterface;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\SDK\Metrics\MeterProviderInterface;
+use OpenTelemetry\SDK\Trace\ReadableSpanInterface;
+use OpenTelemetry\SDK\Trace\SpanProcessor\NoopSpanProcessor;
+use OpenTelemetry\SDK\Trace\SpanProcessorInterface;
+use OpenTelemetry\SemConv\TraceAttributes;
+
+class ResponseTimeSpanProcessor extends NoopSpanProcessor implements SpanProcessorInterface
+{
+    use LogsMessagesTrait;
+    private HistogramInterface $histogram;
+
+    public function __construct(?MeterProviderInterface $meterProvider = null)
+    {
+        $meterProvider = $meterProvider ?? Globals::meterProvider();
+        $meter = $meterProvider->getMeter('sw.apm.request.metrics');
+        $this->histogram = $meter->createHistogram(
+            'trace.service.response_time',
+            'ms',
+            'Duration of each entry span for the service, typically meaning the time taken to process an inbound request.'
+        );
+    }
+
+    /**
+     * Still need the deprecated class constant
+     * @phan-suppress PhanDeprecatedClassConstant
+     */
+    public function onEnd(ReadableSpanInterface $span): void
+    {
+        $parentSpanContext = $span->getParentContext();
+        if ($parentSpanContext->isValid() && !$parentSpanContext->isRemote()) {
+            return;
+        }
+        $durationMs = $span->getDuration() / 1000_000; // ns converts to ms
+        $spanData = $span->toSpanData();
+        $attributes = [
+            'sw.is_error' => $spanData->getStatus()->getCode() === StatusCode::STATUS_ERROR,
+        ];
+        $copy = [TransactionNameSpanProcessor::TRANSACTION_NAME_ATTRIBUTE];
+        if ($span->getKind() === SpanKind::KIND_SERVER) {
+            $copy = array_merge($copy, [
+                TraceAttributes::HTTP_REQUEST_METHOD,
+                TraceAttributes::HTTP_RESPONSE_STATUS_CODE,
```

Comment:
Will upstream instrumentors default this to `0` if status code not available>

Reply:
Upstream by default sets `Status` to `StatusData::unset()` https://github.com/open-telemetry/opentelemetry-php/blob/main/src/SDK/Trace/Span.php#L48. It won't be 0.

Reply:
Thanks! Not sure of best way, but maybe: if it's the return of upstream's `unset` or an invalid code, then don't set the attribute on our metric at all? Or set as zero (APM Python does this).

https://github.com/solarwinds-cloud/trace/blob/master/docs/specs/agent-metrics.md#swo-tags

Reply:
Added the unset test case, the `unset` is mainly determining the `sw.is_error`. Unset is not error, so it will be false.

If I cannot get the span attributes (e.g. HTTP_REQUEST_METHOD, HTTP_STATUS_CODE...), apm-php will not set the attribute on metric at all.

Reply:
Great thank you!